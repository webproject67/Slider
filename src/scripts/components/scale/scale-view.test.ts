import SliderModel from '../model/slider-model';
import ScaleView from './scale-view';

const main = document.createElement('div');
const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  range: 'one',
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'horizontal',
};

const model = new SliderModel(main, state);
const scale = new ScaleView(model);

it('jest snapshots from element', () => {
  expect(scale.element).toMatchSnapshot();
});

it('jest snapshots from new element', () => {
  expect(scale.newElement).toMatchSnapshot();
});

const otherMain = document.createElement('div');
const otherState = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 1,
  range: 'one',
  scale: true,
  step: 2,
  to: -10000,
  toPercent: 100,
  view: 'horizontal',
};

const otherModel = new SliderModel(otherMain, otherState);
const otherScale = new ScaleView(otherModel);

it('jest snapshots from other element', () => {
  expect(otherScale.element).toMatchSnapshot();
});
