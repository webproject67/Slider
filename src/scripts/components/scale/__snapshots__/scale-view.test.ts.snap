// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest snapshots 1`] = `
<div
  class="slider__list"
>
  
          
  <div
    class="slider__item"
    data-testid="scale-horizontal"
  >
    |
            
    <span
      class="slider__item--number slider__item--min"
    >
      0
    </span>
    
          
  </div>
  
        
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  
          
  <div
    class="slider__item"
  >
    |
            
    <span
      class="slider__item--number slider__item--max"
    >
      100
    </span>
    
          
  </div>
  
        
</div>
`;
